### PR TITLE
Validate the expiration date 🗓

### PR DESCRIPTION
**Pull request by ricardopereira**

### CODE REVIEW

**Comment by ricardopereira** [comment]
@dethos PTAL

**Comment by dethos** [comment]
@ricardopereira in this case `datetime.now()` might not always work, given we have users in different timezones. I would suggest the usage of `timezone.now()` from `from django.utils import timezone`.

File: boxes/forms.py [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
 
 
 class CreateBoxForm(ModelForm):
     class Meta:
         model = Box
         fields = ["name", "description", "expires_at"]
 
+    def clean_expires_at(self):
+        # Validate the expiration date
+        if self.expires_at:
```
Comment (ricardopereira) [inline boxes/forms.py]: @dethos The code is failing with: `AttributeError: 'CreateBoxForm' object has no attribute 'expires_at'`. Should I only check the `cleaned_data`?

Comment (dethos) [inline boxes/forms.py]: @ricardopereira yes, you only need to check the cleaned data.

File: boxes/forms.py [hunk 2]
```diff
@@ -1,12 +1,26 @@
...
+            expires_at = self.expires_at
+        else:
+            expires_at = self.cleaned_data.get("expires_at", "")
+
+        if expires_at:
+            # Check if the expiration date is a past date
+            if timezone.now() > expires_at:
```
Comment (dethos) [inline boxes/forms.py]: @ricardopereira in this line you probably will need to do something like this: 

``` python
timezone.localtime(timezone.now(), timezone.get_current_timezone())
```

Sorry I might had mislead you (my fault), but the comparison is only useful if the user timezone is taken into account. `timezone.now()` returns an aware datetime in UTC, so the conversion will be needed.

**Comment by jlbribeiro** [comment]
LGTM, merging :+1:

**Pull request merged**